packages/sdk/src/utils/type-utils.ts

```diff
@@ -18,23 +23,9 @@ export type Split<S extends string | number | symbol, D extends string> = S exte
 
 export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never
 
-type _test_join = Expect<IsEqual<Join<['a', 'b', 'c']>, 'abc'>>
-
-type _test_split = Expect<IsEqual<Split<'a.b.c', '.'>, ['a', 'b', 'c']>>
-
-type _test_union_to_intersection = Expect<
-  IsEqual<
-    UnionToIntersection<
-      | {
-          name: string
-        }
-      | {
-          age: number
-        }
-    >,
-    {
-      name: string
-      age: number
-    }
-  >
->
+/**
+ * removes string index signature from Record
+ */
+export type StrictenRecord<R extends Record<string, any>> = {
+  [K in keyof R as string extends K ? never : K]: R[K]
+}
```

Comment:
*very small nickpick:* perhaps a name like `ToSealedRecord` would be more descriptive of what the type [tries to accomplish](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal)?